qiskit/transpiler/passes/optimization/template_matching/template_substitution.py

```diff
@@ -375,3 +387,89 @@ def run_dag_opt(self):
 
         self.dag_dep_optimized = dag_dep_opt
         self.dag_optimized = dagdependency_to_dag(dag_dep_opt)
+
+    def _fake_bind(self, template_sublist, circuit_sublist):
```

Comment:
"fake_bind" might imply that it doesn't actually bind, but in fact it does bind if some valid params are found. So maybe a different name for this function? something like `_attempt_bind`?

Reply:
I don't think it modifies `self`, though, right? Does that distinction matter here?

Reply:
No, it does not modify `self`. That being said, I like `_attempt_bind` as a bind is not guaranteed.